### PR TITLE
adding VPC Lattice section

### PR DESCRIPTION
**Pull request by federicaciuffo**
#### What this PR does / why we need it:

This PR adds a section on VPC Lattice under the Networking tab.

#### Which issue(s) this PR fixes: NA

Fixes # NA

#### Quality checks

- [x] My content adheres to the style guidelines
- [x] I ran `make test` or `make e2e-test` and it was successful

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: website/docs/networking/vpc-lattice/latticeabtesting.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+1. Verify that `eks-workshop-gw` gateway is created:
+   ```bash
+   $ kubectl get gateway  
+   ```
+   ```
+   $ NAME              CLASS         ADDRESS   READY   AGE
+   $ eks-workshop-gw   aws-lattice                     12min
+   ```
+
+1. Once the gateway is created, find the VPC Lattice service network. Wait until the status is `Reconciled` (this could take about five minutes).
```
Comment (svennam92) [inline website/docs/networking/vpc-lattice/latticeabtesting.md:23]: Can we use a `kubectl wait` here? Or a custom command like here: https://github.com/aws-samples/eks-workshop-v2/blob/main/environment/bin/wait-for-lb (this is used in the Fundamentals load balancer module)

File: website/docs/networking/vpc-lattice/latticeabtesting.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+ Now you can see the associated Service Network created in the VPC console under the Lattice resources.
+![Checkout Service Network](assets/servicenetwork.png)
+
+# Create Routes to targets
+We will show how to preform A/B testing between the two version (`checkout`and `checkoutv2`) using `HTTPRoutes`.
+
+**NOTE:** (At the time of writing, the `targetPort` has to be updated  to `8080` for the Lattice Controller to reroute the traffic). This will be improved in future releases.
+
+```bash
+$ kubectl patch svc checkout -n checkout --patch '{"spec": { "type": "ClusterIP", "ports": [ { "name": "http", "port": 80, "protocol": "TCP", "targetPort": 8080 } ] } }'
```
Comment (svennam92) [inline website/docs/networking/vpc-lattice/latticeabtesting.md:46]: Let's use kustomize rather than patches. It is clearer to understand what is changing using the "diff" tab when using ```kustomization

File: website/docs/networking/vpc-lattice/latticeabtesting.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+$ kubectl patch svc checkout -n checkoutv2 --patch '{"spec": { "type": "ClusterIP", "ports": [ { "name": "http", "port": 80, "protocol": "TCP", "targetPort": 8080 } ] } }' 
+```
+
+1. Create the Kubernetes `HTTPRoute` route that evenly distributes the traffic between `checkout`and `checkoutv2`:
```
Comment (svennam92) [inline website/docs/networking/vpc-lattice/latticeabtesting.md:50]: space between `checkout` and

File: website/docs/networking/vpc-lattice/latticeabtesting.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    parentRef:
+    group: gateway.networking.k8s.io
+    kind: Gateway
+    name: eks-workshop-gw
+```
+
+ Now you can see the associated Service created in the VPC console under the Lattice resources.
+![CheckoutRoute Service](assets/checkoutroute.png)
+
+1. Patch the configmap to point to the new endpoint.
```
Comment (svennam92) [inline website/docs/networking/vpc-lattice/latticeabtesting.md:86]: Numbering reset

File: website/docs/networking/vpc-lattice/cleanup.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
+---
+title: "Cleanup"
+sidebar_position: 25
+---
+
+```bash
```
Comment (svennam92) [inline website/docs/networking/vpc-lattice/cleanup.md:6]: These clean-up steps should be significantly less complex once we switch to using kustomization > sed/patch/etc Kustomize will be able to automatically clean up anything not in the original manifests directory. Also, `reset-environment` is expected to automatically roll back any changes done in a module.

After making above changes to use Kustomize, we will also need to update `reset-environment` with idempotent changes to ensure that users can run `reset-environment` at any stage of running through this lab to reset changes.

File: website/docs/networking/vpc-lattice/setup.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
+---
+title: "VPC Lattice Setup"
+sidebar_position: 10
+---
+
+# Deploying the AWS Gateway API Controller
```
Comment (svennam92) [inline website/docs/networking/vpc-lattice/setup.md:6]: In every other module, we pre-deploy all dependencies to avoid wasting time during in-person labs. However with upcoming changes by @niallthomson to how we are using Terraform and Lattice being a new service, I am okay with including this manual setup flow. I expect most users will want to see the install steps, instead of us abstracting it away from them.

File: website/docs/networking/vpc-lattice/traditionalabtesting.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
+---
+title: "Traditional A/B testing"
```
Comment (svennam92) [inline website/docs/networking/vpc-lattice/traditionalabtesting.md:2]: This is not technically A/B testing, but just deploying a new container. Instead, we could deploy two version of the checkout container with the same labels. This would mean the checkout service would route to both versions of the checkout app, making for rudimentary A/B testing. 2 checkout v1 pods and 2 checkout v2 pods behind the same service should mean 50% to each. Obviously this is not ideal, and sets up nicely for the next step where you can setup granular rules for routing.

**Review by svennam92 — CHANGES_REQUESTED** [review]
Thank you for the PR @federicaciuffo, this is shaping up to be a solid module. I like the A/B testing use-case using Lattice with the checkout service. I've requested some changes to the module before we can merge.

File: website/docs/networking/vpc-lattice/index.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
+---
+title: "Amazon VPC Lattice"
+sidebar_position: 40
+weight: 10
+sidebar_custom_props: {"module": true}
+---
+
+:::danger Read before you start this module
+
+This module currently only works on the [self-hosted version](https://www.eksworkshop.com/docs/introduction/setup/your-account) of eksworkshop as Amazon VPC Lattice resources are not yet supported on [Workshop Studio](https://www.eksworkshop.com/docs/introduction/setup/aws-event).
```
[the remark this answers is not in the document]
Comment (federicaciuffo) [reply website/docs/networking/vpc-lattice/index.md]: Fair, I had the same concern, however Workshop Studio is mentioned in the [introduction of the Workshop](https://www.eksworkshop.com/docs/introduction/setup/aws-event), with a screenshot as well, so people should be somehow familiar with the concept.

File: website/docs/networking/vpc-lattice/setup.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    --cluster=${EKS_CLUSTER_NAME} \
+    --namespace=system \
+    --name=gateway-api-controller \
+    --attach-policy-arn=${VPCLatticeControllerIAMPolicyArn} \
+    --override-existing-serviceaccounts \
+    --region ${AWS_DEFAULT_REGION} \
+    --approve
+```
+
+Run either `kubectl` or `helm` to deploy the controller:
```
[the remark this answers is not in the document]
Comment (federicaciuffo) [reply website/docs/networking/vpc-lattice/setup.md:51]: Added formatting.

File: website/docs/networking/vpc-lattice/setup.md [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+And run `helm` with either install or upgrade
+
+```bash
+$ helm install gateway-api-controller \
+    oci://public.ecr.aws/aws-application-networking-k8s/aws-gateway-controller-chart\
+    --version=v0.0.7 \
+    --set=aws.region=${AWS_DEFAULT_REGION} --set=serviceAccount.create=false --namespace system
+```
+
+Create the `amazon-vpc-lattice` GatewayClass:
```
[the remark this answers is not in the document]
Comment (federicaciuffo) [reply website/docs/networking/vpc-lattice/setup.md:72]: done

File: website/docs/networking/vpc-lattice/traditionalabtesting.md [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+sidebar_position: 15
+---
+
+In this section we will create a new version of the `checkout` service and manually perform A/B testing.
+
+Let's deploy a new version of the `checkout` microservice in a new namespace (`checkoutv2`). We will use kustomize to:
```
[the remark this answers is not in the document]
Comment (federicaciuffo) [reply website/docs/networking/vpc-lattice/traditionalabtesting.md:8]: I tried but this makes the setup (and cleanup) very complex due to the way the application is built. I don't have have a strong preference for this, however this makes it a lot easier.

File: website/docs/networking/vpc-lattice/setup.md (hunk 6 above)
Comment (federicaciuffo) [reply website/docs/networking/vpc-lattice/setup.md:6]: ack

File: website/docs/networking/vpc-lattice/traditionalabtesting.md (hunk 7 above)
Comment (federicaciuffo) [reply website/docs/networking/vpc-lattice/traditionalabtesting.md:2]: As mentioned above, I tried but this makes the setup (and cleanup) very complex due to the way the application is built. I don't have have a strong preference for this, however this makes it a lot easier.

File: website/docs/networking/vpc-lattice/setup.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (federicaciuffo) [reply website/docs/networking/vpc-lattice/setup.md:51]: done

File: website/docs/networking/vpc-lattice/index.md [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+
+Amazon VPC Lattice takes care of common networking tasks such as discovering components, routing  traffic between individual workloads, and authorizing access, eliminating the need for developers to do this themselves through additional software or code. In a few clicks or API calls, developers can configure policies to define how their applications should communicate without any prior networking expertise. 
+
+The primary benefits of using Lattice are:
+
+* Increased developer productivity:  Lattice boosts developer productivity by letting them focus on building features that matters to their business, while it handles networking, security and observability challenges in a uniform way across all compute platforms
+* Improved Security Posture: Lattice  enables  developers  to  easily  authenticate  and  secure  communication  across  applications, without the operational overhead of current mechanisms (e.g. certificate management). With Lattice access policies, developers and cloud admins can enforce granular access control. Lattice can also enforce encryption for traffic in-transit, further increasing security posture
```
Comment (umairishaq) [inline website/docs/networking/vpc-lattice/index.md:29]: Minor sentence case mismatch

File: website/docs/networking/vpc-lattice/index.md [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+* Improved application scalability and resilience: Lattice makes it easy to create a network of deployed applications with rich routing, authentication, authorization, monitoring, and more. Lattice provides all of these benefits with no resource overhead on workloads and can support large scale deployments and many requests per second without adding significant latency. 
+* Deployment flexibility with heterogeneous infrastructure: Lattice  provides  consistent  features  across  all  compute  services – EC2, ECS, EKS, Lambda, and can include services living on-premises, allowing organizations the flexibility to choose the optimal compute infrastructure for their use-case.
+
+The components of Amazon VPC Lattices include:
+
+* **Service network**:
+A shareable, managed logical grouping that contains Services and Policy.
+
+* **Service**:
+Represents an Application Unit with a DNS name and can extend across all compute – instances, containers, serverless. It is made up pf Listener, Target Groups, Target.
```
Comment (umairishaq) [inline website/docs/networking/vpc-lattice/index.md:39]: Minor typo: "It is made up pf Listener"

File: website/docs/networking/vpc-lattice/index.md [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+* Improved Security Posture: Lattice  enables  developers  to  easily  authenticate  and  secure  communication  across  applications, without the operational overhead of current mechanisms (e.g. certificate management). With Lattice access policies, developers and cloud admins can enforce granular access control. Lattice can also enforce encryption for traffic in-transit, further increasing security posture
+* Improved application scalability and resilience: Lattice makes it easy to create a network of deployed applications with rich routing, authentication, authorization, monitoring, and more. Lattice provides all of these benefits with no resource overhead on workloads and can support large scale deployments and many requests per second without adding significant latency. 
+* Deployment flexibility with heterogeneous infrastructure: Lattice  provides  consistent  features  across  all  compute  services – EC2, ECS, EKS, Lambda, and can include services living on-premises, allowing organizations the flexibility to choose the optimal compute infrastructure for their use-case.
+
+The components of Amazon VPC Lattices include:
```
Comment (umairishaq) [inline website/docs/networking/vpc-lattice/index.md:33]: Would it be helpful to add links to docs for various terms/concepts in this section?

File: website/docs/networking/vpc-lattice/gatewayapicontroller.md [hunk 15]
```diff
@@ -0,0 +1,27 @@
+---
+title: "AWS Gateway API Controller"
+sidebar_position: 10
+---
+
+Gateway API is an open-source project managed by the Kubernetes networking community. It is a collection of resources that model application networking in Kubernetes. Gateway API supports resources such as GatewayClass, Gateway, and Route that have been implemented by many vendors and have broad industry support.
+
+Originally conceived as a successor to the well-known Ingress API, the benefits of the Gateway API include (but are not limited to) explicit support for many commonly used networking protocols, as well as tightly integrated support for Transport Layer Security (TLS).
+
+At AWS, we implement the Gateway API to integrate with Amazon VPC Lattice. AWS Gateway API Controller is an open-source project and fully supported by Amazon. When the AWS Gateway API controller is installed in your cluster, the controller watches for the creation of Gateway API resources such as gateways and routes and provisions corresponding Amazon VPC Lattice objects as shown above.
```
Comment (umairishaq) [inline website/docs/networking/vpc-lattice/gatewayapicontroller.md:10]: "At AWS, we implement the Gateway API to integrate with Amazon VPC Lattice." Without some elaboration, for a new reader I sense, a "chicken-and-egg" situation here. "integrate with VPC Lattice" implies existence of "something" but we haven't explained what exists. Would it explain things better if we add some additional details about this integration?

File: website/docs/networking/vpc-lattice/setup.md [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+    --namespace=system \
+    --name=gateway-api-controller \
+    --attach-policy-arn=${VPCLatticeControllerIAMPolicyArn} \
+    --override-existing-serviceaccounts \
+    --region ${AWS_DEFAULT_REGION} \
+    --approve
+```
+
+---
+**CHOOSE AN INSTALLATION METHOD** for the controller between the two options below:
```
Comment (umairishaq) [inline website/docs/networking/vpc-lattice/setup.md:52]: What did we just do here? Would it help to mention at least the key components being created?

File: website/docs/networking/vpc-lattice/setup.md [hunk 17]
```diff
@@ -0,0 +1,75 @@
...
+
+Create the `system` namespace:
+```bash
+$ kubectl apply -f workspace/modules/networking/vpc-lattice/controller/deploy-namesystem.yaml
+```
+Retrieve the policy ARN:
+```bash
+$ export VPCLatticeControllerIAMPolicyArn=$(aws iam list-policies --query 'Policies[?PolicyName==`VPCLatticeControllerIAMPolicy`].Arn' --output text)
+```
+Create an `iamserviceaccount` for pod level permission:
```
Comment (umairishaq) [inline website/docs/networking/vpc-lattice/setup.md:39]: What's the role of this serviceaccount?

File: website/docs/networking/vpc-lattice/latticeabtesting.md [hunk 18]
```diff
@@ -0,0 +1,115 @@
+---
+title: "A/B testing with VPC Lattice"
+sidebar_position: 20
+---
+
+In this section we will show how to use Amazon VPC Lattice to perform A/B testing gradually shifting traffic from the `checkout` service to a new version `checkout-v2`.
```
Comment (umairishaq) [inline website/docs/networking/vpc-lattice/latticeabtesting.md:6]: Adding some more details about the scenario would help orient the user. For example, "we have added a new shipping option as a feature and want to test it" would disclose the intent of the exercise.

File: website/docs/networking/vpc-lattice/latticeabtesting.md [hunk 19]
```diff
@@ -0,0 +1,115 @@
+---
+title: "A/B testing with VPC Lattice"
```
Comment (umairishaq) [inline website/docs/networking/vpc-lattice/latticeabtesting.md:2]: Can we claim this is "A/B testing", as we don't split traffic?

File: website/docs/networking/vpc-lattice/latticeabtesting.md [hunk 20]
```diff
@@ -0,0 +1,115 @@
...
+   reason: Reconciled
+   status: "True"
+   ```
+ Now you can see the associated Service Network created in the VPC console under the Lattice resources.
+![Checkout Service Network](assets/servicenetwork.png)
+
+# Create Routes to targets
+We will show how to preform A/B testing between the two version (`checkout`and `checkoutv2`) using `HTTPRoutes`.
+
+**NOTE:** (At the time of writing, the `targetPort` has to be updated  to `8080` for the Lattice Controller to reroute the traffic). This will be improved in future releases.
```
Comment (umairishaq) [inline website/docs/networking/vpc-lattice/latticeabtesting.md:43]: Leaves the reader really hanging ... what limitation is this? What was the previous value? Other details to educate the reader ...

File: website/docs/networking/vpc-lattice/latticeabtesting.md [hunk 21]
```diff
@@ -0,0 +1,115 @@
...
+    parentRef:
+    group: gateway.networking.k8s.io
+    kind: Gateway
+    name: eks-workshop-gw
+```
+
+ Now you can see the associated Service created in the VPC console under the Lattice resources.
+![CheckoutRoute Service](assets/checkoutroute.png)
+
+3. Patch the configmap to point to the new endpoint.
```
Comment (umairishaq) [inline website/docs/networking/vpc-lattice/latticeabtesting.md:86]: This is the key learning of the whole section. Can we add more details about what is happening under the covers? Without additional details, this does not impart any learning. Adding links to docs would be sufficient too.

File: website/docs/networking/vpc-lattice/cleanup.md (hunk 5 above)
Comment (umairishaq) [reply website/docs/networking/vpc-lattice/cleanup.md:6]: +1 to Sai's take. In any case we should set expectation about the order of these steps and any time delays associated with individual command execution.

File: website/docs/networking/vpc-lattice/setup.md [hunk 22]
```diff
@@ -0,0 +1,75 @@
...
+```bash
+$ aws ecr-public get-login-password --region us-east-1 | helm registry login --username AWS --password-stdin public.ecr.aws
+$ helm install gateway-api-controller \
+    oci://public.ecr.aws/aws-application-networking-k8s/aws-gateway-controller-chart\
+    --version=v0.0.7 \
+    --set=aws.region=${AWS_DEFAULT_REGION} --set=serviceAccount.create=false --namespace system
+```
+---
+
+Before creating a `Gateway`, we need to formalize the types of load balancing implementations with a [GatewayClass](https://gateway-api.sigs.k8s.io/concepts/api-overview/#gatewayclass).
```
Comment (umairishaq) [inline website/docs/networking/vpc-lattice/setup.md:70]: This is one of the key module concepts. Adding some details about "types of load balancing implementations" would help connect the details of how things work. Link to existing docs would serve well too.

File: website/docs/networking/vpc-lattice/latticeabtesting.md
Comment (umairishaq) [inline website/docs/networking/vpc-lattice/latticeabtesting.md:1]: This is the only file/section that uses numbered steps. I like it but it feels a bit inconsistent compared to other files. Can we change to one style for consistency?

File: website/docs/networking/vpc-lattice/setup.md [hunk 23]
```diff
@@ -0,0 +1,75 @@
...
+First, configure security group to receive traffic from the VPC Lattice fleet. You must set up security groups so that they allow all Pods communicating with VPC Lattice to allow traffic on all ports from the `169.254.171.0/24` address range. See [Control traffic to resources using security groups](https://docs.aws.amazon.com/vpc/latest/userguide/VPC_SecurityGroups.html) for details. You can use the following managed prefix to provide the values:
+
+```bash
+$ PREFIX_LIST_ID=$(aws ec2 describe-managed-prefix-lists --query "PrefixLists[?PrefixListName=="\'com.amazonaws.$AWS_DEFAULT_REGION.vpc-lattice\'"].PrefixListId" | jq --raw-output .[])
+$ MANAGED_PREFIX=$(aws ec2 get-managed-prefix-list-entries --prefix-list-id $PREFIX_LIST_ID --output json  | jq -r '.Entries[0].Cidr')
+$ CLUSTER_SG=$(aws eks describe-cluster --name eks-workshop --output json| jq -r '.cluster.resourcesVpcConfig.clusterSecurityGroupId')
+$ aws ec2 authorize-security-group-ingress --group-id $CLUSTER_SG --cidr $MANAGED_PREFIX --protocol -1
+```
+
+Create a policy (`recommended-inline-policy.json`) in IAM with the following content that can invoke the gateway API and copy the policy arn for later use:
```
Comment (umairishaq) [inline website/docs/networking/vpc-lattice/setup.md:19]: `"Resource": "*"` typically spook customers. Should we add a line about it being a exercise and that for real-world to use appropriate policy?

**Review by umairishaq — COMMENTED** [review]
Really great work! Added a few comments ...

File: website/docs/networking/vpc-lattice/setup.md (hunk 17 above)
Comment (federicaciuffo) [reply website/docs/networking/vpc-lattice/setup.md:39]: it's the IRSA setup for the controller

File: website/docs/networking/vpc-lattice/latticeabtesting.md (hunk 20 above)
Comment (federicaciuffo) [reply website/docs/networking/vpc-lattice/latticeabtesting.md:43]: Hi Umair, for the moment this is a bug of the controller: https://github.com/aws/aws-application-networking-k8s/issues/86.

File: website/docs/networking/vpc-lattice/latticeabtesting.md (hunk 21 above)
Comment (federicaciuffo) [reply website/docs/networking/vpc-lattice/latticeabtesting.md:86]: Agree, added a `tip` section to highlight this part.

File: website/docs/networking/vpc-lattice/traditionalabtesting.md [hunk 24]
```diff
@@ -0,0 +1,73 @@
+---
+title: "Traditional A/B testing"
+sidebar_position: 15
+---
+
+In this section we will create a new version of the `checkout` service and manually perform A/B testing.
```
[the remark this answers is not in the document]
Comment (federicaciuffo) [reply website/docs/networking/vpc-lattice/traditionalabtesting.md:6]: done

File: website/docs/networking/vpc-lattice/setup.md (hunk 22 above)
Comment (federicaciuffo) [reply website/docs/networking/vpc-lattice/setup.md:70]: changed

File: website/docs/networking/vpc-lattice/setup.md (hunk 23 above)
Comment (federicaciuffo) [reply website/docs/networking/vpc-lattice/setup.md:19]: done

File: website/docs/networking/vpc-lattice/traditionalabtesting.md [hunk 25]
```diff
@@ -0,0 +1,73 @@
...
+Let's deploy a new version of the `checkout` microservice in a new namespace (`checkoutv2`). We will use kustomize to: 
+
+* Change the namespace to `checkoutv2`, 
+* Substitute the image with a new version,
+* Create a new version of `checkout`.
+
+```bash
+$ CHECKOUT_IMG="public.ecr.aws/aws-containers/retail-store-sample-checkout:latest"
+$ CHECKOUT_V2="public.ecr.aws/y1b0a4i8/checkoutv2:latest"
+$ sed -i 's/checkout/checkoutv2/g' workspace/manifests/checkout/kustomization.yaml
```
[the remark this answers is not in the document]
Comment (federicaciuffo) [reply website/docs/networking/vpc-lattice/traditionalabtesting.md]: changed to work with kustomize

File: website/docs/networking/vpc-lattice/index.md (hunk 12 above)
Comment (federicaciuffo) [reply website/docs/networking/vpc-lattice/index.md:29]: can you clarify this for me?

File: website/docs/networking/vpc-lattice/latticeabtesting.md
Comment (federicaciuffo) [reply website/docs/networking/vpc-lattice/latticeabtesting.md:1]: done

File: website/docs/networking/vpc-lattice/index.md (hunk 14 above)
Comment (federicaciuffo) [reply website/docs/networking/vpc-lattice/index.md:33]: added link to aws docs

File: website/docs/networking/vpc-lattice/cleanup.md (hunk 5 above)
Comment (federicaciuffo) [reply website/docs/networking/vpc-lattice/cleanup.md:8]: Simplified the setup/cleanup using kustomize as suggested. I’m not changing the part where we patch the UI configmap and the existing service because it would be a little bit of an overkill. Please, check the new version. With the current cleanup steps the reset-environment should bring everything back to normal from a user point of view.

File: website/docs/networking/vpc-lattice/latticeabtesting.md (hunk 20 above)
Comment (umairishaq) [reply website/docs/networking/vpc-lattice/latticeabtesting.md:43]: Sorry Federica, my intent was not clear here. My suggestion is to give the reader an idea what this comment means with a little explanation and linking to the issue. Without it, the reader can be puzzled by what's going on.

File: website/docs/networking/vpc-lattice/setup.md (hunk 17 above)
Comment (federicaciuffo) [reply website/docs/networking/vpc-lattice/setup.md:39]: understood, changed.

File: website/docs/networking/vpc-lattice/latticeabtesting.md (hunk 20 above)
Comment (federicaciuffo) [reply website/docs/networking/vpc-lattice/latticeabtesting.md:43]: changed

File: website/docs/networking/vpc-lattice/traditionalabtesting.md [hunk 26]
```diff
@@ -0,0 +1,66 @@
...
+sidebar_position: 15
+---
+
+
+We have modified the `checkout` microservice adding a feature that provides an additional shipping option. In this section, we want to test this new feature performing A/B testing. 
+
+Let's deploy a new version of the `checkout` microservice in a new namespace (`checkoutv2`) using `kustomize`.
+
+```bash
+$ kubectl apply -k workspace/modules/networking/vpc-lattice/abtesting/
```
Comment (svennam92) [inline website/docs/networking/vpc-lattice/traditionalabtesting.md:12]: Instead of creating a new namespace for v2 and patching the config, can we simply just create another deployment pointing to the new image? We can use a different name (to avoid conflict), but the same labels. This allows both versions of the pod to be served by the same service. This is rudimentary traffic routing, with a 50/50 chance of hitting v1 vs v2. This also removes the need to patch the config.

Optionally, you can then delete the v1 deployment afterwards. This allows cleanup to be much easier as well.

**Review by svennam92 — CHANGES_REQUESTED** [review]
We're getting closer, but I'll reiterate that we should never uses patches — please replace them with kustomize manifests. This is a critical design choice of the workshop and required before we can merge.

File: website/docs/networking/vpc-lattice/cleanup.md [hunk 27]
```diff
@@ -0,0 +1,39 @@
+---
+title: "Cleanup"
```
Comment (svennam92) [inline website/docs/networking/vpc-lattice/cleanup.md:2]: I don't see any changes to reset-environment. User should be able to run the reset-environment script at any point of this workshop to undo these changes. This is required before we can merge.

File: website/docs/networking/vpc-lattice/cleanup.md [hunk 28]
```diff
@@ -0,0 +1,39 @@
...
+
+```bash
+$ kubectl delete -f ./workspace/modules/networking/vpc-lattice/routes
+$ kubectl delete -f ./workspace/modules/networking/vpc-lattice/controller/eks-workshop-gw.yaml 
+$ kubectl delete -f ./workspace/modules/networking/vpc-lattice/controller/gatewayclass.yaml 
+```
+---
+Delete the AWS Gateway API Controller. 
+
+**THIS STEP DEPENDS FROM THE INSTALLATION METHOD YOU CHOOSE**
```
Comment (svennam92) [inline website/docs/networking/vpc-lattice/cleanup.md:16]: Workshop attendees are not expected to run clean up steps manually. They should be able to simply run `reset-environment`. This page should be accurate for reference, but should be automated into the `bin/reset-environment` script.

File: website/docs/networking/vpc-lattice/latticeabtesting.md [hunk 29]
```diff
@@ -0,0 +1,119 @@
...
+    name: eks-workshop-gw
+```
+
+ Now you can see the associated Service created in the VPC console under the Lattice resources.
+![CheckoutRoute Service](assets/checkoutroute.png)
+
+Patch the configmap to point to the new endpoint.
+
+```bash
+$ export CHECKOUT_ROUTE_DNS='http://'$(kubectl get httproute checkoutroute -n checkout -o json | jq -r '.status.parents[0].conditions[0].message' | cut  -c 11-)
```
Comment (svennam92) [inline website/docs/networking/vpc-lattice/latticeabtesting.md:88]: is the checkout route static?

File: environment/workspace/modules/networking/vpc-lattice/abtesting/deploymentv2.yaml [hunk 30]
```diff
@@ -0,0 +1,11 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: checkout
+spec:
+  template:
+    spec:
+      containers:
+        - name: checkout
+          image: "public.ecr.aws/y1b0a4i8/checkoutv2:latest"
```
Comment (svennam92) [inline environment/workspace/modules/networking/vpc-lattice/abtesting/deploymentv2.yaml:10]: this image should be built as part of the upstream repo: https://github.com/aws-containers/retail-store-sample-app

File: website/docs/networking/vpc-lattice/latticeabtesting.md (hunk 29 above)
Comment (federicaciuffo) [reply website/docs/networking/vpc-lattice/latticeabtesting.md:88]: No, is dynamically generated by Lattice .

File: environment/workspace/modules/networking/vpc-lattice/abtesting/deploymentv2.yaml (hunk 30 above)
Comment (federicaciuffo) [reply environment/workspace/modules/networking/vpc-lattice/abtesting/deploymentv2.yaml]: fixed using new env variable provided by @niallthomson
